google/calendar/server.py

```diff
@@ -0,0 +1,717 @@
+              new_calendar_id: Annotated[str, Field(description="calendar id to move the event to")],
+              cred_token: str = None) -> dict:
+    """Moves an event to a different google calendar."""
+    if calendar_id == "":
+        raise ValueError("argument `calendar_id` can't be empty")
+    if event_id == "":
+        raise ValueError("argument `event_id` can't be empty")
+    if new_calendar_id == "":
+        raise ValueError("argument `new_calendar_id` can't be empty")
+    service = get_client(cred_token)
+
+    try:
+        existing_event = (
+            service.events().get(calendarId=calendar_id, eventId=event_id).execute()
+        )
+        if (
+            existing_event_type := existing_event.get("eventType")
+        ) not in MOVABLE_EVENT_TYPES:
+            raise ValueError(
+                f"Events with type '{existing_event_type}' can not be moved."
+            )
+
+        event = (
+            service.events()
+            .move(calendarId=calendar_id, eventId=event_id, destination=new_calendar_id)
+            .execute()
+        )
+        return event
+    except HttpError as err:
+        raise ToolError(f"Failed to move event {event_id} to calendar {new_calendar_id}. HttpError: {err}")
+    except Exception as e:
+        raise ToolError(f"Failed to move event {event_id} to calendar {new_calendar_id}. Exception: {e}")
+
+
+@mcp.tool(
+    exclude_args=["cred_token"],
+)
+def quick_add_event(text: Annotated[str, Field(description="The text of the event to add")],
+                    calendar_id: Annotated[str, Field(description="The ID of the calendar to add event for")] = "primary",
+                    cred_token: str = None) -> dict:
+    """Quickly adds an event to the calendar based on a simple text string."""
+    if text == "":
+        raise ValueError("argument `text` can't be empty")
+    service = get_client(cred_token)
+
+    try:
+        event = service.events().quickAdd(calendarId=calendar_id, text=text).execute()
+        return event
+    except HttpError as err:
+        raise ToolError(f"Failed to quick add event to calendar {calendar_id}. HttpError: {err}")
+    except Exception as e:
+        raise ToolError(f"Failed to quick add event to calendar {calendar_id}. Exception: {e}")
+
+
+@mcp.tool(
+    exclude_args=["cred_token"],
+)
+def create_event(calendar_id: Annotated[str, Field(description="Calendar id to create event in. Set to `primary` to create event in the primary calendar")],
+                 summary: Annotated[str, Field(description="Event title")] = "My Event",
+                 location: Annotated[str, Field(description="Geographic location of the event as free-form text.")] = "",
+                 description: Annotated[str, Field(description="Event description")] = "",
+                 time_zone: Annotated[str | None, Field(description="Event time zone to create. Defaults to the user's timezone. Must be a valid IANA timezone string")] = None,
```

Comment:
> `Default's to the user's timezone.`

Is this still true when this runs a as a standalone MCP server?

Reply:
It'd be server time unless we do something with `OBOT_USER_TIMEZONE`

Reply:
not really, now it defaults to the server timezone

Reply:
I hope Obot can somehow supply this information to the MCP server though